io/io/src/RRawFileWin.cxx

```diff
@@ -0,0 +1,75 @@
+// @(#)root/io:$Id$
+// Author: Jakob Blomer
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "ROOT/RRawFileWin.hxx"
+
+#include <cerrno>
+#include <cstddef>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <stdexcept>
+#include <string>
+
+
+ROOT::Detail::RRawFileWin::RRawFileWin(const std::string &url, ROOT::Detail::RRawFile::ROptions options)
+  : ROOT::Detail::RRawFile(url, options)
+  , fileptr(nullptr)
+{
+}
+
+ROOT::Detail::RRawFileWin::~RRawFileWin()
+{
+   if (fileptr != nullptr)
+      fclose(fileptr);
+}
+
+void ROOT::Detail::RRawFileWin::Seek(long offset, int whence)
+{
+   int res = fseek(fileptr, offset, whence);
+   if (res != 0)
+      throw std::runtime_error("Cannot seek in '" + fUrl + "', error: " + std::string(strerror(errno)));
+}
+
+size_t ROOT::Detail::RRawFileWin::DoPread(void *buffer, size_t nbytes, std::uint64_t offset)
+{
+   EnsureOpen();
+   Seek(offset, SEEK_SET);
+   size_t res = fread(buffer, 1, nbytes, fileptr);
+   if ((res < nbytes) && (ferror(fileptr) != 0)) {
+      clearerr(fileptr);
+      throw std::runtime_error("Cannot read from '" + fUrl + "', error: " + std::string(strerror(errno)));
```

Comment:
I would recommend against throwing on errors as it will be 'slower' and more combursome for the user to handle (especially since this function, for example, is not exception safe [it has partially changed the state of the object])

Reply:
that is indeed up for debate but we are both provide user level interface (unlikely to check results) and framework level interface (likely to check the results, and RDataFrame is a framework using this interface).   I would venture that the error reporting would be better implemented in the RDataFrame layer.